python/cuml/feature_extraction/_tfidf.py

```diff
@@ -30,7 +32,15 @@ def _sparse_document_frequency(X):
         return cp.diff(X.indptr)
 
 
-class TfidfTransformer:
+def _get_dtype(X):
+    """
+        Returns the valid dtype for tf-idf transoformer
+    """
+    dtype = X.dtype if X.dtype in FLOAT_DTYPES else cp.float32
```

Comment:
Can you copy in the `FLOAT_DTYPES` directly from sklearn?

Aside from tests and benchmarks, we have been trying to keep sklearn dependencies out of the Python.

Reply:
Resolved by https://github.com/rapidsai/cuml/pull/2698/commits/e6033f7e06e0930a6b497dd4fa8b326e41fc33b0